### PR TITLE
Adding Jakarta Messaging 3.0 support

### PR DESCRIPTION
**Pull request by samueleresca**
See: #483 

Few notes regarding the PR after spending some time on this:

- Jakarta JMS 3.1 requires Java 11+, see [Jakarta Messaging 3.1.0 Release Review](https://projects.eclipse.org/projects/ee4j.messaging/reviews/3.1.0-release-review). This PR is upgrading to [Jakarta JMS 3.0](https://projects.eclipse.org/projects/ee4j.jms/reviews/3.0.0-release-review) which is the latest release supporting Java 8.
- ActiveMQ Classic is used in the jms connector to perform the integration tests. ActiveMQ classic cannot be upgraded to a version later than `5.16.7` because the later versions don't support Java 8. 
- ActiveMQ Classic 5.16.7 only [supports Javax JMS 1.1](https://activemq.apache.org/components/classic/download/).
- I replaced ActiveMQ Classic with ActiveMQ Artemis, which [supports both the new versions of JMS and also Java 8](https://activemq.apache.org/components/artemis/download/past_releases). I wasn't able to port the following 2 tests using ActiveMQ Artemis:
   - `JmsProducer should retry sending on network failures`
   - `The JMS Transactional Connectors ensure no message loss or starvation when exceptions occur in a stream missing commits`

### CODE REVIEW

File: jakarta/src/main/scala/org/apache/pekko/stream/connectors/jakarta/SendRetrySettings.scala
Comment (pjfanning) [inline jakarta/src/main/scala/org/apache/pekko/stream/connectors/jakarta/SendRetrySettings.scala:14]: Could you use jakartams? Jakarta covers a lot more than messaging.

File: jakartams/src/test/scala/org/apache/pekko/stream/connectors/jakartams/EmbeddedActiveMQServer.scala
Comment (samueleresca) [inline jakartams/src/test/scala/org/apache/pekko/stream/connectors/jakartams/EmbeddedActiveMQServer.scala:1]: Using the Apache header as this is has been written by me.

File: .github/workflows/check-build-test.yml
Comment (pjfanning) [inline .github/workflows/check-build-test.yml:106]: minor format issue

File: build.sbt
Comment (pjfanning) [inline build.sbt:277]: could this be `lazy val jakartams` ?

File: jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala
Comment (pjfanning) [inline jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala:18]: Not a big deal but if you `import org.apache.pekko`, can the other pekko imports omit the `org.apache` bit?

I won't block a merge over this.

File: project/Dependencies.scala
Comment (pjfanning) [inline project/Dependencies.scala:352]: can you pin this in `.scala-steward.conf` ?

File: project/Dependencies.scala
Comment (pjfanning) [inline project/Dependencies.scala:354]: could you create an Artemis Version constant?

File: jakartams/src/test/scala/org/apache/pekko/stream/connectors/jakartams/scaladsl/CachedConnectionFactory.scala
Comment (pjfanning) [inline jakartams/src/test/scala/org/apache/pekko/stream/connectors/jakartams/scaladsl/CachedConnectionFactory.scala:34]: could remove this comment

File: jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala
Comment (laglangyue) [inline jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala:36]: This API is intended for users to use, and usually user level parameters should be fully named instead of using `value`. The same goes for other places, such as `c` for config

File: jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala
Comment (laglangyue) [inline jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala:50]: It's a Double? In conf, it seems a Long

File: jakartams/src/main/resources/reference.conf
Comment (laglangyue) [inline jakartams/src/main/resources/reference.conf:17]: the comment of max-backoff is same to backoff-factor

Comment (samueleresca) [reply jakartams/src/main/resources/reference.conf:17]: Updated the comment in both jms and jakartams

File: jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala
Comment (samueleresca) [reply jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala]: @laglangyue  Thanks for spotting this. I have updated please let me know if my understanding is right.

>  The same goes for other places, such as c for config

I'm not sure I follow. What's `c` for config are you referring to?

File: jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala
Comment (samueleresca) [reply jakartams/src/main/scala/org/apache/pekko/stream/connectors/jakartams/ConnectionRetrySettings.scala]: As a factor it can be fractional (e.g.: https://github.com/apache/pekko-connectors/blob/main/jms/src/main/resources/reference.conf#L73)

**Comment by pjfanning** [comment]
@samueleresca is this PR complete? I'm hoping to start a discussion about a Pekko Connectors 1.1.0-M1 release. This can be delayed until after that release if that suits you better.

**Comment by samueleresca** [comment]
> @samueleresca is this PR complete? I'm hoping to start a discussion about a Pekko Connectors 1.1.0-M1 release. This can be delayed until after that release if that suits you better.

There is a comment @laglangyue that I need to address, apart from that it is ready. It would be great to merge this for Pekko Connectors 1.1.0-M1.

**Comment by pjfanning** [comment]
@samueleresca This is the last PR blocking the release. Again, I'm happy to move on with the M1 release without this and to have it merged prior to the full 1.1.0 release.

**Comment by samueleresca** [comment]
> @samueleresca This is the last PR blocking the release. Again, I'm happy to move on with the M1 release without this and to have it merged prior to the full 1.1.0 release.

@pjfanning The PR is ready apart few minor comments that I think shouldn't block the merge. So I'm happy to get the approval and get this merged

**Review by pjfanning — APPROVED** [review]
lgtm

**Pull request merged**
